feature: update documentation format

ju-Skinner opened the pull request:
# Description

* @pine-ds/doc-components:  Introduce new component library for Documentation
* Refactor existing documentation mdx files to use new file format and components
* 

Fixes #(issue)

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

- [X] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests you've added and run to verify your changes.
Provide instructions so that we can reproduce.
Please also list any relevant details for your test configuration.

- [ ] unit tests
- [ ] e2e tests
- [ ] accessibility tests
- [ ] tested manually
- [ ] other:

**Test Configuration**:

- Pine versions:
- OS:
- Browsers:
- Screen readers:
- Misc:

# Checklist:

If not applicable, leave options unchecked.

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing tests pass locally with my changes
- [ ] Design has QA'ed and approved this PR

ju-Skinner committed:
docs(pds-sortable): update docs.mdx to use new format and raw mdx file

docs(sortable): add playground to storybook docs.mdx file

docs(sortable): fix code samples to follow JSX syntax

Commit:
docs(chip): update docs to new format

ju-Skinner committed:
docs(pds-radio): create new mdx file structure for doc site

docs(radio): add playground to storybook docs.mdx file

docs(radio): fix code samples to follow JSX syntax

ju-Skinner committed:
docs(pds-avatar): update docs.mdx to use new format and raw mdx file

docs(pds-avatar): refactor to add playground section

QuintonJason committed:
docs(pds-link): update docs.mdx to use new format and raw mdx file

docs(pds-link): refactor to add playground section

docs(link): refactor to add playground section

QuintonJason committed:
docs(pds-divider): update docs.mdx to use new format and raw mdx file

docs(divider): refactor to add playground section

QuintonJason committed:
docs(pds-image): update docs.mdx to use new format and raw mdx file

docs(pds-image): refactor to add playground section

docs(image): fix source example to JSX syntax

QuintonJason committed:
docs(pds-input): update docs.mdx to use new format and raw mdx file

docs(pds-input): refactor to add playground section

docs(input): fix code samples to follow JSX syntax

ju-Skinner committed:
docs(pds-progress): create new mdx file structure for doc site

docs(progress): add playground to storybook docs.mdx file

docs(progress): fix code samples to follow JSX syntax

Commit:
docs(checkbox): update docs format and add playground

Commit:
docs(switch): updates component docs with new layout

Commit:
docs(copytext): update docs format and add playground

Commit:
docs(pds-tooltip): update docs.mdx to use new format and raw mdx file

docs(tooltip): remove unused imports

Commit:
docs(textarea): update docs format and add playground

ju-Skinner committed:
chore: move @pine-ds/doc-compents from dependencies to devDependencies

Commit:
docs(button): update docs format and add playground

ju-Skinner committed:
docs(pds-icon): refactor mdx files to allow for use outside of storybook

ju-Skinner committed:
docs(pds-tabs): refactor mdx files to allow for use outside of storybook

pixelflips commented on libs/core/src/components/pds-image/docs/pds-image.mdx:
```diff
@@ -0,0 +1,68 @@
+<DocCanvas client:only
+  mdxSource={{
+    react: `
+<pds-image
```
I think this should be `PdsImage`

pixelflips commented on libs/core/src/components/pds-input/docs/pds-input.mdx:
```diff
@@ -0,0 +1,121 @@
+
+<DocCanvas client:only
+  mdxSource={{
+    react: '<PdsInput componentId="pds-input-error-example" label="Name" type="email" error-message="Please provide a valid email address" value="Frank Dux"></PdsInput>',
```
I think this should be `errorMessage`.

pixelflips commented on libs/core/src/components/pds-progress/docs/pds-progress.mdx:
```diff
@@ -0,0 +1,81 @@
+Displays the `percent` value as text beside the progress.
+
+<DocCanvas client:only mdxSource={{
+  react: `<PdsProgress componentId="show-percent" label="Show percent" percent="50" show-percent={true}></PdsProgress>`,
```
I think this should be `showPercent`

pixelflips commented on libs/core/src/components/pds-progress/docs/pds-progress.mdx:
```diff
@@ -0,0 +1,81 @@
+Allows for `pds-progress` to be filled with a custom color. Accepts any valid color token or CSS color value.
+
+<DocCanvas client:only mdxSource={{
+  react: `<PdsProgress componentId="fillcolor" label="Fill color" percent="95" fill-color="#86D5BC"></PdsProgress>`,
```
I think this should be `fillColor`

pixelflips commented on libs/core/src/components/pds-progress/docs/pds-progress.mdx:
```diff
@@ -0,0 +1,81 @@
+
+<DocCanvas client:only mdxSource={{
+  react: `<PdsTooltip content="50%" has-arrow="false" placement="top" style={{ width: '100%' }}>
+    <PdsProgress component-id="tooltip-demo" label="Progress w/ tooltip" percent="50"></PdsProgress>
```
`componentId` here.

pixelflips commented on libs/core/src/components/pds-progress/docs/pds-progress.mdx:
```diff
@@ -0,0 +1,81 @@
+    <PdsProgress component-id="tooltip-demo" label="Progress w/ tooltip" percent="50"></PdsProgress>
+  </PdsTooltip>`,
+  webComponent: `<pds-tooltip content="50%" has-arrow="false" placement="top" style={{ width: '100%' }}>
+    <pds-progress component-id="tooltip-demo" label="Progress w/ tooltip" percent="50"></pds-progress>
```
`componentId` here.

pixelflips commented on libs/core/src/components/pds-radio/docs/pds-radio.mdx:
```diff
@@ -0,0 +1,94 @@
+Message is a string of text that displays below the label text.
+
+<DocCanvas client:only mdxSource={{
+  react: `<PdsRadio componentId="message1" label="Label" helper-message="This is short message text." />`,
```
`helperMessage` here.

pixelflips commented on libs/core/src/components/pds-radio/docs/pds-radio.mdx:
```diff
@@ -0,0 +1,94 @@
+An optional error message can be displayed for additional instructions, displayed independently from the helper message.
+
+<DocCanvas client:only mdxSource={{
+  react: `<PdsRadio componentId="invalid" label="Label" invalid error-message="This is a short error message" />`,
```
`errorMessage` here.

pixelflips commented on libs/core/src/components/pds-sortable/docs/pds-sortable.mdx:
```diff
@@ -0,0 +1,301 @@
+## Properties
+
+<DocArgsTable componentName='pds-sortable' />
+
```
nitpick, extra space.

pixelflips commented on libs/core/src/components/pds-sortable/docs/pds-sortable.mdx:
```diff
@@ -0,0 +1,301 @@
+When `border` is set to `true`, the component will render a border around all items.
+
+<DocCanvas client:only mdxSource={{
+  react:`<PdsSortable border component-id="bordered">
```
`componentId` here

pixelflips commented on libs/core/src/components/pds-sortable/docs/pds-sortable.mdx:
```diff
@@ -0,0 +1,301 @@
+The `sortable-item-actions` slot is provided to place content within the `pds-sortable-item__actions` element as children, but this slot is only rendered if the `enable-actions` property is set to `true`.
+
+<DocCanvas client:only mdxSource={{
+  react: `<PdsSortable component-id="actions">
```
`componentId` here

pixelflips commented on libs/core/src/components/pds-sortable/docs/pds-sortable.mdx:
```diff
@@ -0,0 +1,301 @@
+
+<DocCanvas client:only mdxSource={{
+  react: `<PdsSortable component-id="actions">
+  <PdsSortableItem enable-actions>
```
`enableActions` here and the following items below.

pixelflips commented on libs/core/src/components/pds-sortable/docs/pds-sortable.mdx:
```diff
@@ -0,0 +1,301 @@
+
+<DocCanvas client:only mdxSource={{
+  react: `<PdsSortable border={true} dividers={true} componentId="demo">
+  <PdsSortableItem enable-actions handle>
```
`enableActions` here and the following items below.

pixelflips commented on libs/core/src/components/pds-progress/stories/pds-progress.docs.mdx:
```diff
@@ -3,73 +3,8 @@ import { html, render } from 'lit';
 
```
I think progress is missing the playground

pixelflips commented on libs/core/src/components/pds-radio/stories/pds-radio.docs.mdx:
```diff
@@ -3,86 +3,8 @@ import { html, render } from 'lit';
 
```
i think radio is missing the playground

pixelflips commented on libs/core/src/components/pds-tooltip/docs/pds-tooltip.mdx:
```diff
@@ -0,0 +1,116 @@
+    react: `
+      <PdsTooltip hasArrow="true" placement="right">
+        text
+        <div slot="content">
+          <p>this is a sentence in a tooltip.</p>
+          <p>this is a sentence in a tooltip.</p>
+        </div>
```
Can prob be handled in a follow up but looks like the HTML content is inheriting some colors from somewhere, which is odd...

![Screenshot 2023-12-06 at 4 42 48 PM](https://github.com/Kajabi/pine/assets/1175111/d9eea555-e972-4224-a1d1-e7725692b98e)

ju-Skinner replied:
Here is the style that is affecting it. I'm just not sure as to why. Looking into this now
![image](https://github.com/Kajabi/pine/assets/1633290/af85df0c-719c-4b6d-8427-898b9cff7566)

ju-Skinner replied:
You can see removing color, resolves the issue
![image](https://github.com/Kajabi/pine/assets/1633290/49ddc048-bbde-499d-9b7c-987bae28d5c8)

pixelflips replied:
seems odd that what I am assuming are the storybook styles would affect/override the component styles.

pixelflips reviewed (changes requested):
Overall I dig this approach. A few comments that are mostly related to typos and some missing playground sections.

If the playground sections are intentional just mark those comments as resolved and ignore.

ju-Skinner committed:
chore: fix package-lock due to NX missing platform dependency

ju-Skinner committed:
refactor: component mdx files to assign value to the docSource property

Read the doc.json from dist folder and pass the components attribute to docSource property

ju-Skinner committed:
chore: replace storybook logo with Pine logo

ju-Skinner committed:
chore: replace favicon details with new pine icon

ju-Skinner committed:
feat: create react doc-components library

feat(react-syntax-highlighter): add react-syntax-highlighter

refactor(docArgsTable): parse mdx so it can be rendered as html

ju-Skinner committed:
chore(core): update React dev dependency to match react and doc-components

pixelflips reviewed (approved):
Outside of the non-blocking previously mentioned tooltip color issue, this is awesome! LGTM!

anechol reviewed (approved):
Love it

ju-Skinner committed:
chore: update start command to include doc-components

QuintonJason reviewed (approved):
The docs are working and the stencil components are being pulled in as well. Great work 🏆

ju-Skinner committed:
refactor(docArgsTable): changed from table element to css table and flex grid

ju-Skinner committed:
refactor(docCanvas): style enhancements

ju-Skinner committed:
docs(doc-components): readme documentation

ju-Skinner committed:
chore: change the entry points for publishing

ju-Skinner committed:
chore: package updates
